### PR TITLE
[RFC] Add an option to use numbered registers for smaller deletions #8121

### PR DESCRIPTION
**Pull request by Himura2la**
Trying to implement #8121 (Add an option to use numbered registers for smaller deletions)

### CODE REVIEW

File: src/nvim/option_defs.h [hunk 1]
```diff
@@ -581,6 +581,7 @@ EXTERN int p_sm;                // 'showmatch'
 EXTERN int p_smd;               // 'showmode'
 EXTERN long p_ss;               // 'sidescroll'
 EXTERN long p_siso;             // 'sidescrolloff'
+EXTERN int p_sdt;               // 'smalldeletethreshold'
```
Comment (justinmk) [inline src/nvim/option_defs.h]: Not sure about this option name, it's unusually long. 'delcombine' is the only existing option from which precedent can be drawn...

**Comment by justinmk** [comment]
Please add a note a `:help quote-`. 

> I did it global instead of buffer-specific for simplicity. Maybe I could make it buffer-specific

Global is fine, let's not overkill it.

**Comment by Himura2la** [comment]
I remember about the docs, but I really think it should be buffer-local. Because it's especially useful for plain text with very long lines and not so useful for code.
I've got a problem with this item 
> //   - Add a BV_XX or WV_XX entry to option_defs.h
I found no any usage of `BV_*` items from the enum, how and why should I add my option there?

File: src/nvim/option_defs.h (hunk 1 above)
Comment (Himura2la) [reply src/nvim/option_defs.h]: I'll think about it

**Comment by justinmk** [comment]
Well, that's another field that now exists on every buffer. And almost no one is going to use this option.

**Comment by Himura2la** [comment]
OK, that was a kind of challenge ))
Reverting a commit is easier than creating it.
So you believe it should be global, right? By the way, what's wrong with another field? Does it influence the performance or memory?

What about naming, I'm thinking of `bigdelmin`. Sounds strange, though. What do you think?

**Comment by justinmk** [comment]
IMO it should be global until it's proven that people actually care enough to want otherwise.

> What about naming, I'm thinking of bigdelmin

That name makes no sense to me.

**Comment by Himura2la** [comment]
OK, I'll leave it global. Now I know that global options are more preferable.
`set smalldel=2` ?
It's a quite hard concept to express without using 'big' or 'large' words (because there's no such concepts as 'large deletion' or 'big deletion') and the word 'threshold' because it's too wide...

**Comment by Himura2la** [comment]
@justinmk I think I'm ready. Thanks for your help.

**Comment by justinmk** [comment]
@Himura2la Is there a reason the `TextYankPost` event can't be used to achieve what you want?

**Comment by Himura2la** [comment]
Maybe it is possible to write an autocommand for this, if I knew about this mechanism earlier, this could be my way.
From my point of view, the autocommand's disadvantages:
- Large code in `.vimrc` for a simple task
- Hard to find, I've seen [a similar solution](https://vi.stackexchange.com/a/2511) and I don't want to put something similar into my `.vimrc`. I try to follow the way of mastering Vim without tons of customization.

I just noticed the absence of such option strange. And found those who also think so.

**Comment by bfredl** [comment]
@justinmk The issue is that `TextYankPost` would need to do the cycling of the numbered registers manually. I suppose generally usable would be to add `setreg(X, text, kind)`, where X is a new value meaning 'set register "1, also shift the remaining numbered registers'.

**Comment by sorennikolaus** [comment]
I'd really like to have this merged.

**Comment by justinmk** [comment]
Only thing blocking this is a better option name. Some sort of "flags" option similar to 'wildoptions'. E.g. for https://github.com/neovim/neovim/pull/11530 we added 'jumpoptions'. 

Perhaps this could be a flag in 'cpoptions'.

**Comment by Himura2la** [comment]
Sorry for being inactive, I really missed the last comment about naming, and checking my PRs just now.
If we call it `cpoptions`, then, I guess, it should accept not a number? So what's your idea?

**Comment by danielo515** [comment]
This is a really good improvement. Please go ahead

**Comment by Himura2la** [comment]
I guess, we need more ideas on naming...

**Comment by danielo515** [comment]
> I guess, we need more ideas on naming...

Are there any specific requirements? Like, should it include the word option?

**Comment by Himura2la** [comment]
delcombine -- existing option. source of inspiration
smalldeletethreshold -- most descriptive, but too long
smalldel -- LGTM, I'm not sure what @justinmk think about it

**Comment by Himura2la** [comment]
delcombine -- existing option. source of inspiration
smalldeletethreshold -- most descriptive, but too long
smalldel -- LGTM, I'm not sure what @justinmk think about it

**Comment by pete-may** [comment]
I'd definitely like to have this feature.

**Comment by bfredl** [comment]
As `TextYankPost` now exists this can be handled by plugins.

**Pull request closed without merging**